crates/http/src/layers/catch_http_codes.rs

```diff
@@ -45,10 +45,16 @@ impl<S, E> Error<S, E> {
     }
 }
 
+/// A layer that catches responses with the HTTP status codes lying within
+/// `bounds` and then maps the requests into a custom error type using `mapper`.
 #[derive(Clone)]
 pub struct CatchHttpCodes<S, M> {
+    /// The inner service
     inner: S,
+    /// Which HTTP status codes to catch
     bounds: (Bound<StatusCode>, Bound<StatusCode>),
+    /// The function used to convert errors, which must be `Clone` and
+    /// `Fn(Response<ResBody>) -> E + Send + Clone + 'static`.
```

Comment:
I think `M: Fn(Response<ResBody>) -> E + Send + Clone + 'static` should be understood as

```
M: Fn(Response<ResBody>) -> E
M: Send
M: Clone
M: 'static
```

so you're mentioning Clone twice here?